### DISCUSSION TITLE
Is it possible to exclude specific versions from `matchUpgradeType`-like  rules?

### DISCUSSION THREAD

**Discussion by aisamu** [renovatebot/renovate#17571] [Request Help]
To avoid the XY problem, here's the full context:

- We disable auto merge and require dashboard approvals for major changes on our default configuration:
  ```
  "major": {
    "automerge": false,
    "dependencyDashboardApproval": true,
  }
  ```

- When we receive a notification for a major change, we inspect Changelogs and add backwards/forward compatible fixes to our repos ahead of the PR.
- Since we have a lot of repositories, such changes are written once and propagated via [codemods](https://github.com/facebookarchive/codemod) or [projen](https://github.com/projen/projen).
  
Once the fix is in place everywhere, we'd like "bless" that version and have renovate automatically create those PRs.

Essentially, we're trying to express:

> **For this particular package + version, skip the dashboard approval**

Is that achievable, or is it a doomed pursuit?

The literal/naive version had the side-effect of disabling all updates to “non-blessed” versions:
  ```
    // bless target-package@3.x
    {
      "dependencyDashboardApproval": false,
      "matchPackageNames": [
        "target-package"
      ],
      "allowedVersions": "/^3/"
    },
  ```
NB: there’s currently no support for combining `matchUpgradeType` and `allowedVersions` into the same rule.

**Comment by PhilipAbed** [comment]
To my knowledge, the last rule that matches the dependency will overrule any other matches before, 
did you try to do this extra `bless package rule` after the `match major rule`? see if it overrides it?

**Reply by aisamu** [reply]
The `major` rule lives on a shared renovate configuration. The "blessing" rule was added directly on an arbitrary repo.
I'm assuming local rules always go after inherited defaults, so I believe it overrode the original `major` top-level rule.
Another evidence is that it disabled all other updates to the package.

**Reply by PhilipAbed** [reply]
i've tested, and it works fine if the order is correct PSB:
![image](https://user-images.githubusercontent.com/42116482/187956932-b9243e77-16f3-4707-8a1c-c7a2df73a683.png)


so what you are trying to do is possible, and you are doing it the correct way.

**Reply by aisamu** [reply]
Note that this not fully represent the original intention:
> **For this particular package and this particular version, skip the dashboard approval**

IIUC, the added rule above means

> **For this particular package, for all its versions, skip the dashboard approval**

In terms more specific to your example, I'm aiming at getting `Update dependency commander to v9` to go through without approval (as you've shown), but I'd still expect a future `Update dependency commander to v10` to require approval.

**Reply by PhilipAbed** [reply]
Then u should use the dashboard checkbox instead of creating a specific rule... otherwise u will.need to add the rule.. then remove the rule.. where u couldve just went to issues-> dashboard and hit the checkbox then trigger run again...

**Reply by aisamu** [reply]
>  u should use the dashboard checkbox

You've hit the nail on the head: that's how we're currently operating and precisely what we're trying to solve in an automated fashion.

We have *hundreds* of repositories being managed by renovate - manually clicking the checkbox on each dashboard, besides being a tad inefficient, is proving to be error-prone/unreliable!

**Reply by PhilipAbed** [reply]
Maybe instead of setting the allowedVersions value to `^3` u set it to also to ` <= the upgraded version ` everytime.. so what u want will happen..

If i wanted to automate this i would grab the dependendency dashboard from issues using the github api and then updating it with the [x] checkbox 

Hope those ideas help cuz im out of ideas.
renovate doesnt support such automation currently

**Reply by aisamu** [reply]
> renovate doesnt support such automation currently

Understood!

> Hope those ideas help

Regardless, thanks a lot for your time!
